lib/integration.js

```diff
@@ -0,0 +1,229 @@
+    const editorPath = textEditor.getPath()
+      if (!editorPath) {
+        return
+      }
+
+      handleErrors(editorPath);
+
+      const changeSubscription = textEditor.onDidSave(function(event){
+        handleErrors(event.path);
+      })
+      subscriptions.add(changeSubscription)
+
+      const destroySubscription = textEditor.onDidDestroy(function() {
+        subscriptions.remove(destroySubscription)
+        subscriptions.remove(changeSubscription)
+        linter.setMessages(editorPath, [])
+      })
+
+      subscriptions.add(destroySubscription)
+    }));
+  }
+
+  handleErrors(_path)
+  {
+    if(_path != null) {
+      Linterhub.analyzeFile(_path).then(function(data){
+        linter.setMessages(_path, data);
+      });
+    }
+    else {
+      Linterhub.analyze().then(function(data){
+        linter.setAllMessages(data);
+      });
+    }
+  }
+
+  /**
+  * Warning callback
+  *
+  * @param {Object} event - Link to event node ---------------------
+  */
+  warningCallback(event) {
+    // let linter = event.target.getAttribute('linter-name');
+    let _warning = event.target.getAttribute('rule-name');
+    if (_warning === 'null') _warning = null;
+    let _file = event.target.getAttribute('file-name');
+    if (_file === 'null') _file = null;
+    let _line = event.target.getAttribute('line-number');
+    if (_line === 'null') _line = null;
+    let target = event.currentTarget;
+    Linterhub.ignoreWarning({
+      file: _file, line: _line, error: _warning,
+    }).then(() => {
+      this.handleErrors(_file);
+    });
+  }
+
+  /**
+  * @param {Object} settings - Linterhub settings
+  */
+  saveConfig(settings) {
+    console.log("Saving config..");
+    atom.config.set('linterhub-atom.mode', settings.linterhub.mode);
+    atom.config.set('linterhub-atom.path_cli', settings.linterhub.cliPath);
+  }
+
+  /**
+  * @param {String} path - Current project path
+  * @return {String} - Return curent project path
+  */
+  normalizePath(path) {
+    return path;
+  }
+
+  /**
+  * @param {Object} data -
+  * @return {Object} -
+  */
+  sendDiagnostics(data) {
+    let json = JSON.parse(data);
+    let results = [];
+    for (let index = 0; index < json.length; index++) {
+      let linterResult = json[index];
+      linterResult.Model.Files.forEach((file) => {
+        file.Errors.forEach((error) => {
+          results.push(this.convertError(error, linterResult.Name, file.Path));
+        });
+      });
+    }
+    return results;
+  }
+
+  /**
+  * @param {Object} message -
+  * @param {Object} linter -
+  * @param {Object} file -
+  * @return {Object} -
+  */
+  convertError(message, linter, file) {
+    let severity = 'warning';
+    switch (Number(message.Severity)) {
+      case 0:
+        severity = 'error';
+        break;
+      case 1:
+        severity = 'warning';
+        break;
+      case 2:
+        severity = 'info';
+        break;
+      case 3:
```

Comment:
You can just move this right under `case 2:` since both cases have the same effect.